vignettes/NSSP_nowcast.Rmd

```diff
@@ -0,0 +1,510 @@
+
+```{r}
+max_delay <- 25
+nowcast_date <- max(long_df$reference_date) - days(30)
+```
+
+## Format for `baselinenowcast`
+We'll start by generating a reporting triangle from the `syn_nssp_rep_tri_df`, after we've removed all of the reports from after the nowcast date.
+We will need to fill in all the possible reference dates and report dates with delays up until the maximum delay.
+We'll do this by first making a "spine" of all of the date combinations we need, and then joining the dataset to that spine.
+As a final step, we'll pivot filter out the report date column and pivot from wide to long format.
+
+
+```{r}
+training_df <- long_df |>
+  filter(report_date <= nowcast_date) |>
+  mutate(delay = as.integer(report_date - reference_date)) |>
+  filter(delay <= max_delay) |>
+  select(reference_date, report_date, count)
+
+rep_tri_df <- expand_grid(
+  reference_date = seq(min(long_df$reference_date),
+    nowcast_date,
+    by = "day"
+  )
+) |>
+  rowwise() |>
+  reframe(
+    reference_date = reference_date,
+    report_date = seq(reference_date, min(
+      reference_date + max_delay,
+      nowcast_date
+    ),
+    by = "day"
+    )
+  ) |>
+  mutate(delay = as.integer(report_date - reference_date)) |>
+  left_join(training_df,
+    by = c("reference_date", "report_date")
+  ) |>
+  replace_na(list(count = 0)) |>
+  select(-report_date) |>
+  pivot_wider(
+    names_from = delay,
+    values_from = count
+  )
+head(rep_tri_df)
+```
+As a final pre-processing step, we'll format the reporting triangle as just a matrix, for compatibility with the low-level functions in `baselinenowcast`.
+```{r}
+rep_tri <- rep_tri_df |>
+  select(-reference_date) |>
+  as.matrix()
+```
+## Specify the `baselinenowcast` model
+
+We will use the helper function `allocate_reference_times()` to determine how many reference times to use for delay estimation and uncertainty estimation.
+We will use the defaults for this function, which uses 3 times the maximum delay as the total amount of training volume, split evenly between reference times used for delay estimation and reference times used for producing retrospective nowcasts for uncertainty estimation.
+In this case, this will result in 37 reference times used for delay estimation, and 38 used for uncertainty estimation.
+```{r}
+ref_time_allocation <- allocate_reference_times(rep_tri)
+n_history_delay <- ref_time_allocation$n_history_delay
+n_retrospective_nowcasts <- ref_time_allocation$n_retrospective_nowcasts
+```
+
+# Run the `baselinenowcast` workflow
+Here we demonstrate the workflow using the intermediate-level functions. For more details, see the [Getting Started](baselinenowcast.html) vignette.
+
+First, we generate the point estimate of the nowcast as a "filled in" reporting triangle
+```{r}
+pt_nowcast_matrix <- estimate_and_apply_delay(rep_tri, n = n_history_delay)
+```
+
+Next, we estimate and apply uncertainty by generating retrospective nowcasts and comparing against later observed data.
```

Comment:
tell me about or link (ideally link) to some of the basic assumptions being made here (and above)